### PR TITLE
timerfd api for freebsd 14.

### LINKED ISSUE #3339: timerfd on FreeBSD >= 14.0
Same API as Linux. See [timerfd(2)](https://man.freebsd.org/cgi/man.cgi?query=timerfd&sektion=2&manpath=FreeBSD+15.0-CURRENT) and [`<sys/timerfd.h>`](https://github.com/freebsd/freebsd-src/blob/main/sys/sys/timerfd.h) for details.

Similar to `eventfd` the ABI is new, so no need to gate by `cfg(freebsd14)`.

### PR DESCRIPTION
**Pull request by devnexen**
close #3339

### CODE REVIEW

File: src/unix/bsd/freebsdlike/freebsd/mod.rs [hunk 1]
```diff
@@ -5406,6 +5411,15 @@ extern "C" {
         infotype: *mut ::c_uint,
         flags: *mut ::c_int,
     ) -> ::ssize_t;
+
+    pub fn timerfd_create(clockid: ::clockid_t, flags: ::c_int) -> ::c_int;
```
Comment (asomers) [inline src/unix/bsd/freebsdlike/freebsd/mod.rs:5415]: Nit: the header and man page both define `clockid` as `int`, not `clockid_t`, though those are the same type.  Is the difference intentional?

Comment (devnexen) [reply src/unix/bsd/freebsdlike/freebsd/mod.rs]: no I just misread.

**Review by asomers — APPROVED** [review]

**Review by JohnTitor — APPROVED** [review]

**Pull request merged**